pkg/controllers/allocation/binpacking/packer.go

```diff
@@ -63,6 +81,11 @@ type Packing struct {
 // It follows the First Fit Decreasing bin packing technique, reference-
 // https://en.wikipedia.org/wiki/Bin_packing_problem#First_Fit_Decreasing_(FFD)
 func (p *packer) Pack(ctx context.Context, schedule *scheduling.Schedule, instances []cloudprovider.InstanceType) []*Packing {
+	startTime := time.Now()
+	defer func() {
```

Comment:
is there a reason to wrap this in a func rather than just doing: 

```
startTime := time.Now()
defer packTimeHistogram.Observe(time.Since(startTime).Seconds())
```

Reply:
According to [Defer, Panic, and Recover](https://go.dev/blog/defer-panic-and-recover) from the Go blog: 
> A deferred function’s arguments are evaluated when the defer statement is evaluated.

So the proposed code will not record the elapsed time because only the `.Observe()` call is defered; `time.Since(startTime).Seconds()` will be evaluated immediately.